packages/manager/apps/dedicated/client/app/ip/components/list/export-csv/ip-ip-export-csv.controller.js

```diff
@@ -1,106 +1,118 @@
-import keys from 'lodash/keys';
+import {
+  CSV_HEADERS,
+  CSV_FILENAME,
+  CSV_DATA_ENCODING,
+  CSV_DATA_SCHEME,
+  CSV_SEPARATOR,
+} from './ip-ip-export.constants';
 import { TRACKING_PREFIX } from '../list.constant';
 
 export default /* @ngInject */ (
   $scope,
-  $q,
   $translate,
   Alerter,
   atInternet,
+  Ip,
 ) => {
-  let timeoutObject = null;
-
-  $scope.data = $scope.currentActionData.ipsList;
+  let cancelFetch = null;
 
   $scope.loading = {
     exportCsv: false,
   };
 
-  function printCsv(datas) {
-    let csvContent = '';
-    let dataString;
-    let link;
-    let headers;
-    let fileName;
+  function fetchIps() {
+    const { serviceType, otherParams } = $scope.currentActionData;
+    const { cancel, request } = Ip.fetchIps({
+      serviceType,
+      otherParams,
+      pageNumber: 1,
+      pageSize: 5000,
+    });
+    return {
+      cancel,
+      request: request.then(({ ips }) => ips),
+    };
+  }
 
-    if (datas && timeoutObject) {
-      // get column name
-      headers = datas.headers;
-      csvContent += `${headers.join(';')}\n`;
-      angular.forEach(datas.ips, (data, index) => {
-        dataString = '';
-        angular.forEach(headers, (header) => {
-          dataString += `${data[header]};`;
-        });
-        csvContent += index < datas.ips.length ? `${dataString}\n` : dataString;
-      });
+  function formatIps(ips) {
+    return ips.map((ip) =>
+      CSV_HEADERS.reduce(
+        (object, { key, getValue }) => ({ ...object, [key]: getValue(ip) }),
+        {},
+      ),
+    );
+  }
 
-      fileName = 'export_ips.csv';
-      link = document.createElement('a');
-      if (link.download !== undefined) {
-        link.setAttribute(
-          'href',
-          `data:text/csv;charset=ISO-8859-1;base64,${btoa(csvContent)}`,
-        );
-        link.setAttribute('download', fileName);
-        link.style = 'visibility:hidden';
-        document.body.appendChild(link);
-        link.click();
-        document.body.removeChild(link);
-      } else {
-        window.open(
-          `data:text/csv;charset=ISO-8859-1;base64,${btoa(csvContent)}`,
-        );
-      }
+  function formatDataURL(ips) {
+    const content = [
+      CSV_HEADERS.map(({ key }) => key).join(CSV_SEPARATOR.CELL),
+      ...ips.map((ip) => Object.values(ip).join(CSV_SEPARATOR.CELL)),
+    ].join(CSV_SEPARATOR.ROW);
+    return [
+      CSV_DATA_SCHEME,
+      Buffer.from(content).toString(CSV_DATA_ENCODING),
+    ].join(CSV_SEPARATOR.SCHEME);
+  }
 
-      Alerter.success($translate.instant('ip_export_csv_success'));
-    } else if (!datas) {
-      Alerter.error($translate.instant('ip_export_csv_error'));
+  function downloadCSV(dataURL) {
+    const link = document.createElement('a');
```

Comment:
i guess it's a good practice to use `$document` instead of `document`.
It will avoid to manage manualy and synchronize manualy the digest cycle.

 ```suggestion
    const link = $document[0].createElement('a');
```